### ISSUE TITLE
[Linux desktop] Failure to compile on 32-bit systems

### ISSUE THREAD

**Issue by PureTryOut** [organicmaps/organicmaps#2859]
Due to https://github.com/organicmaps/organicmaps/blob/master/generator/intermediate_data.cpp#L131

Is 32-bit actively unsupported? Wouldn't that lock out a lot of Android phones as well? Or is it somehow trying to compile code which it just shouldn't do?

**Comment by biodranik** [comment]
It's generator only. You do not need to build it to view maps. Build only `desktop` target.

**Comment by PureTryOut** [comment]
Great, how do I do that? :sweat_smile:  Reading the `CMakeLists.txt` I see nothing obvious related to it, only a way to _not_ build the `desktop` target. Also, what is the generator exactly? Is it something I should want anyway on 64-bit?

**Comment by biodranik** [comment]
Did you check docs in the repo?

**Comment by PureTryOut** [comment]
I have, I did not find anything there related to building/skipping the generator. I'm probably just blind though.

The docs talk about using `tools/unix/build_omim.sh` but I'm using CMake directly. I did compare it's method of calling CMake to my own and I don't believe I'm missing any useful parameters.

**Comment by biodranik** [comment]
After running cmake you run the build with ninja or make. For them, you can specify the target `desktop`.

**Comment by PureTryOut** [comment]
Yeah I noticed that and tried it, but the desktop target is unknown sadly (btw you can also do `cmake --build <build directory> --target <target, e.g. desktop>`).

**Comment by biodranik** [comment]
I see, there is a wrong check in CMakeLists. Please fix and test it.

**Comment by vng** [comment]
So, with this patch of root CMakeLists.txt :
```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index ce3f1b5fa8..3d667788e2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -88,7 +88,7 @@ add_compile_options(-ffast-math)
 
 if (${CMAKE_BUILD_TYPE} STREQUAL "Debug")
   add_definitions(-DDEBUG -fno-omit-frame-pointer)
-elseif (${CMAKE_BUILD_TYPE} MATCHES "^Rel")
+elseif (${CMAKE_BUILD_TYPE} MATCHES "Rel")
   add_definitions(-DRELEASE)
   add_compile_options(-Ofast)
 else()
```
Next command works fine for me:
cmake -B build -DCMAKE_BUILD_TYPE=MinSizeRel .
